tests/integration/workflows/test_workflows_serializers.py

```diff
@@ -0,0 +1,158 @@
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from flask import json
+
+from invenio_accounts.testutils import login_user_via_session
+
+from invenio_workflows import workflow_object_class
+
+from inspire_utils.record import get_value
+from inspire_schemas.api import load_schema, validate
+
+
+def test_workflow_json_serializer_does_not_resolve_if_fuzzy_matches_is_missing(workflow_app):
+    obj = workflow_object_class.create(
+        data=_get_valid_workflow_metada(),
+        data_type='hep',
+        id_user=1,
+    )
+    obj.extra_data = {
+        'matches': {}
+    }
+    obj.save()
+    with workflow_app.test_client() as client:
+        login_user_via_session(client, email='cataloger@inspirehep.net')
+        response = client.get('/api/holdingpen/' + str(obj.id))
+        result = json.loads(response.data)
+        resolved_fuzzy = get_value(result, '_extra_data.matches.fuzzy')
+        assert resolved_fuzzy is None
+
+
+def test_workflow_json_serializer_resolves_fuzzy_matches(workflow_app):
+    _insert_matched_hep_record_with_validation(1234, 'title', 'abstract')
+
+    obj = workflow_object_class.create(
+        data=_get_valid_workflow_metada(),
+        data_type='hep',
+        id_user=1,
+    )
+    obj.extra_data = {
+        'matches': {
+            'fuzzy': [
+                {'$ref': 'https://localhost:5000/api/literature/1234'}
+            ]
+        }
+    }
+    obj.save()
+    expected_fuzzy = [{
+        'control_number': 1234,
+        'abstract': 'abstract',
+        'title': 'title'
+    }]
+    with workflow_app.test_client() as client:
+        login_user_via_session(client, email='cataloger@inspirehep.net')
+        response = client.get('/api/holdingpen/' + str(obj.id))
+        result = json.loads(response.data)
+        resolved_fuzzy = result['_extra_data']['matches']['fuzzy']
+        assert resolved_fuzzy == expected_fuzzy
+
+
+def test_workflow_json_serializer_resolves_fuzzy_matches_without_abstract(workflow_app):
+    _insert_matched_hep_record_with_validation(1234, 'title')
+
+    obj = workflow_object_class.create(
+        data=_get_valid_workflow_metada(),
+        data_type='hep',
+        id_user=1,
+    )
+    obj.extra_data = {
+        'matches': {
+            'fuzzy': [
+                {'$ref': 'https://localhost:5000/api/literature/1234'}
+            ]
+        }
+    }
+    obj.save()
+    expected_fuzzy = [{
+        'control_number': 1234,
+        'title': 'title'
+    }]
+    with workflow_app.test_client() as client:
+        login_user_via_session(client, email='cataloger@inspirehep.net')
+        response = client.get('/api/holdingpen/' + str(obj.id))
+        result = json.loads(response.data)
+        resolved_fuzzy = result['_extra_data']['matches']['fuzzy']
+        assert resolved_fuzzy == expected_fuzzy
+
+
+def _insert_matched_hep_record_with_validation(control_number, title, abstract_value=None):
```

Comment:
hmm I wouldn't add more logic to the tests, you can create fixtures for each case or, even better, run on your test case

Reply:
maybe fixture is better, because when this goes inside tests, it makes them pretty unreadable.